tachyon/c/zk/plonk/halo2/kzg_family_prover_impl.h

```diff
@@ -32,16 +36,25 @@ class KZGFamilyProverImpl : public ProverImplBase<PS> {
   using ProverImplBase<PS>::ProverImplBase;
 
   CProjectivePoint* CommitRaw(const std::vector<ScalarField>& scalars) const {
+#if TACHYON_CUDA
+    return DoMSM(this->pcs_.GetDeviceG1PowersOfTau(), scalars);
+#else
     return DoMSM(this->pcs_.GetG1PowersOfTau(), scalars);
+#endif
   }
 
   CProjectivePoint* CommitLagrangeRaw(
       const std::vector<ScalarField>& scalars) const {
+#if TACHYON_CUDA
+    return DoMSM(this->pcs_.GetDeviceG1PowersOfTauLagrange(), scalars);
```

Comment:
Please check if gpu is configured to be run like how we do in KZG code. 

```suggestion
    if (this->pcs_.UsesGPU()) {
      return DoMSM(this->pcs_.GetDeviceG1PowersOfTauLagrange(), scalars);
    }  
```